delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,520 @@
+
+        # Phone and / or mobile should only be displayed if instruction to
+        # Notify delivery by telephone is set
+        if partner.postlogistics_notification == "email":
+            recipient["email"] = partner.email
+        elif partner.postlogistics_notification == "phone":
+            recipient["phone"] = partner_phone
+            if partner_mobile:
+                recipient["mobile"] = partner_mobile
+        elif partner.postlogistics_notification == "sms":
+            recipient["mobile"] = partner_mobile
+
+        return recipient
+
+    def _prepare_customer(self, picking):
+        """ Create a ns0:Customer as a dict from picking
+
+        This is the Postlogistic Customer, thus the sender
+
+        :param picking: picking browse record
+        :return a dict containing data for ns0:Customer
+
+        """
+        company = picking.company_id
+        partner = company.partner_id
+
+        customer = {
+            "name1": partner.name,
+            "street": partner.street,
+            "zip": partner.zip,
+            "city": partner.city,
+            "country": partner.country_id.code,
+            "domicilePostOffice": picking.carrier_id.postlogistics_office or None,
+        }
+        logo = picking.carrier_id.postlogistics_logo
+        if logo:
+            logo_image = Image.open(BytesIO(base64.b64decode(logo)))
+            logo_format = logo_image.format
+            customer["logo"] = logo.decode()
+            customer["logoFormat"] = logo_format
+        return customer
+
+    def _get_label_layout(self, picking):
+        """
+        Get Label layout define in carrier
+        """
+        return picking.carrier_id.postlogistics_label_layout.code
+
+    def _get_output_format(self, picking):
+        """
+        Get Output format define in carrier
+        """
+        return picking.carrier_id.postlogistics_output_format.code
+
+    def _get_image_resolution(self, picking):
+        """
+        Get Output Resolution Code define in carrier
+        """
+        return picking.carrier_id.postlogistics_resolution.code
+
+    def _get_license(self, picking):
+        """ Get the license
+
+        Take it from carrier and if not defined get the first license.
+
+        :return: license number
+        """
+        franking_license = picking.carrier_id.postlogistics_license_id
+        return franking_license.number
+
+    def _prepare_attributes(self, picking, pack=None, pack_num=None, pack_total=None):
+        packaging = (
+            pack
+            and pack.packaging_id
+            or picking.carrier_id.postlogistics_default_packaging_id
+        )
+        services = packaging._get_packaging_codes()
+
+        total_weight = pack.shipping_weight if pack else picking.shipping_weight
+        total_weight *= 1000
+
+        if not services:
+            raise exceptions.UserError(
+                _("No Postlogistics packaging services found in this picking.")
```

Comment:
```suggestion
                _("No PostLogistics packaging services found in packaging {} of picking {}.").format(...)
```

Instead of picking please mention the picking and the packaging, plus put the names in there.
It will greatly help with module that will create labels in batch to be able to identify which packaging has an issue.

We should not assume the label is always generated from the UI of the picking.

We can probably also mention the `Basic services` as it requires at least a Basic service (like ECO, PRI....) and then can have additional services.